packages/my-account/src/components/composite/Address/CustomerAddressForm/index.tsx

```diff
@@ -13,20 +14,31 @@ import {
 
 import { AddressInputGroup } from "#components/composite/Address/AddressInputGroup"
 import Title from "#components/ui/Title"
-import CustomerAddressContext from "#context/CustomerAddressContext"
+import { appRoutes } from "#data/routes"
+import { AppContext } from "#providers/AppProvider"
+import { CustomerAddressContext } from "#providers/CustomerAddressProvider"
 
-interface Props {
-  onClose: () => void
-}
-
-function CustomerAddressForm({ onClose }: Props): JSX.Element {
+function CustomerAddressForm(): JSX.Element | null {
+  const appCtx = useContext(AppContext)
+  const ctx = useContext(CustomerAddressContext)
   const { t } = useTranslation()
-  const { address, setShowAddressForm } = useContext(CustomerAddressContext)
+  const [showForm, setShowForm] = useState(false)
+  const [, setLocation] = useLocation()
+  const address = ctx?.address
+  const [, params] = useRoute(appRoutes.editAddress.path)
+  const addressId = params == null ? undefined : params.addressId
+
+  useEffect(() => {
+    if (address != null || addressId === undefined) setShowForm(true)
+    return () => {
+      setShowForm(false)
+    }
+  }, [address])
 
-  return (
+  return showForm === false ? null : (
```

Comment:
Shouldn't `showForm` be removed? Shouldn't be the page to rendering it?

Reply:
You are right! We added it yesterday with Ale to check the hydration but in the end is no more needed because I added the check for current path thanks to the value of `addressId` param. Removed!